thunder/__init__.py

```diff
@@ -814,7 +814,50 @@ def maybe_call_epilogue(cache_entry, result, pro_to_epi):
 
         return result
 
+    def unwrap_inner_exception(c: Callable) -> Callable:
+        def _thunder_unwrap_inner_exception(*args, **kwargs):
+            # Run the function, and caputre the exception if there is one.
+            try:
+                return c(*args, **kwargs)
+            except InnerException as e:
+                exc = e.value
+
+            def internal_to_thunder(co):
+                if co is thunder_general_jit.__code__ or co is _thunder_unwrap_inner_exception.__code__:
+                    return True
+                return co.co_filename.endswith("thunder" + os.sep + "core" + os.sep + "interpreter.py") and (
+                    co.co_name in ("fn_", "fn_2")
+                )
```

Comment:
- Why just fn_ and fn_2 and not all of the methods in the interpreter?
- Please don't use the filename/code object here. Instead define `_THUNDER_MODULE_NAME = 'thunder.core.interpreter'` in interpreter.py and detect whether that is in f_globals of the frame.

If we drop all interpreter methods, we likely want to keep internal frames at the top of the stacktrace.

Reply:
Using `f_globals` instead of `co_name` and excluding everything in `thunder.core.interpreter` sounds good.

As for the ones that use code objects, those live elsewhere, in files that I don't want to exclude.

Reply:
Which functions we want to filter does not have an f_globals with `__name__`?